src/mlpack/methods/lmnn/lmnn_function_impl.hpp

```diff
@@ -107,35 +223,65 @@ double LMNNFunction<MetricType>::Evaluate(const arma::mat& transformation)
         // Calculate cost due to {data point, target neighbors, impostors}
         // triplets.
         double eval = 0;
-        if ((iteration - 1 % range == 0) && (iteration - 1 != 0))
+
+        // Bounds for eval.
+        if (transformationOld.n_elem != 0 && evalOld(l, j, i) < -1)
```

Comment:
Perhaps ``.is_empty()`` is more descriptive?

Reply:
Done!